### ISSUE TITLE
Support additional dependencies for mojo executions aka tasks in Scala DSL

### ISSUE THREAD

**Issue by lefou** [takari/maven-polyglot#32]
Currently, there is no option to add extra dependencies. The only options are:
* id
* phase
* profileId

**Comment by lefou** [comment]
@jvanzyl Any pointers, e.g. to the relevant Java API that handles the dependencies/classpath for the execution? You said over IRC, that the Java API supports additional dependencies. Could you point me to an example, or show an example in one of the other scripting languages where this is already possible?

Are there any workarounds, e.g. adding the required dependency to the project dependencies or something like this?

**Comment by jvanzyl** [comment]
In the pom.xml dependencies are added for plugins not goals (or mojo executions). So you should be able to add to the equivalent of the plugin element. Am I understanding correctly that you want to add some dependencies that contain implementations your Mojo needs to execute?

**Comment by lefou** [comment]
Yes, your understanding is correct. I want to use some API contributed by some additional dependencies in the script file. Those dependencies are not necessarily also dependencies of the project sources.

Regarding the goal vs. plugin part. How can I access the plugin that corresponds to the Scala Tasks DSL? The Scala Tasks DSL internally delegates to a class of type `ExecuteManager`. How exactly should it be used to add the dependencies?

**Comment by jvanzyl** [comment]
I would have to look at the Scala DSL, I'm not all that familiar with it. You might want to ask @huntc.

**Comment by jvanzyl** [comment]
@lefou Also we have our Maven Dev Hangout today at 4pm if you want to chat about anything Polyglot related. I think you're in Europe and might be a bit late for you but here's the info if you're interested:

http://takari.io/events.html

**Comment by huntc** [comment]
@lefou I may not fully appreciate what you're asking for. You receive an `org.sonatype.maven.polyglot.execute.ExecuteContext` for your code block:

https://github.com/takari/maven-polyglot/blob/master/polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/model/Task.scala

The tasks are registered via polyglot here:

https://github.com/takari/maven-polyglot/blob/master/polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/ScalaModelReader.scala#L143

There's thus nothing special concerning the Scala APIs here.

**Comment by lefou** [comment]
Hm, not sure how we can move forward from here. Beside the original title and description, the issue is, that there is no way to use additional dependencies in the Scala-based pom file per se. Without this ability, we are not able to use extra API in the script. Can somebody tell if the `pom.scala` is evaluated line-by-line or compiled at once. In the latter case I don't see a way to add extra dependencies by doing it in the pom.scala file itself. Where is the best place to discuss ideas?

**Comment by huntc** [comment]
@lefou Are you attending ScalaDays in Amsterdam by any chance? I'm going to be there... we can discuss.

**Comment by lefou** [comment]
No, I'm not going there this year. But still like to discuss it. Not sure, if I have the resources to implement what's needed though. I'm on IRC (e.g. #scala, #maven) and Gitter for a chat.

**Comment by mosabua** [comment]
I am inclined to close this issue. Wdyt @lefou and @huntc ?

**Comment by lefou** [comment]
@mosabua Please keep to open. Maybe, I can speak with @huntc this year at ScalaDays in Berlin.
